core/src/main/java/org/lflang/generator/python/PythonGenerator.java

```diff
@@ -366,8 +365,8 @@ public boolean isOSCompatible() {
   public void doGenerate(Resource resource, LFGeneratorContext context) {
     // Set the threading to false by default, unless the user has
     // specifically asked for it.
-    if (!targetConfig.setByUser.contains(TargetProperty.THREADING)) {
-      targetConfig.threading = false;
+    if (!targetConfig.threading.isSet()) {
```

Comment:
I agree that it's weird, but I've seen many such enigmatic workarounds sprinkled throughout the code base, without any proper explanation, too... I tried as much as possible for my changes to remain meaning-preserving, which is why I didn't address this. I don't think that `initialValue` should take a `Target` as a parameter because that information generally isn't available upon initialization...

Reply:
Interestingly enough, the failing test passes when I remove this code :shrug: